tools/assetpipelineeditor/camerawidget.cpp

```diff
@@ -30,6 +30,33 @@
 #include "ui_camerawidget.h"
 #include "ui_vectorwidget.h"
 
+#include <QFileDialog>
+#include <QFileInfo>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QSettings>
+
+namespace {
+    const QLatin1String LASTPATHSETTING("mainwindow/lastPath");
+
+    QJsonObject toJson(const QVector3D &v)
```

Comment:
vector3DToJson() ? or you need to have a templated T fromJson(const QJsonObject) below